forem/forem
# Use generalized profile fields for left sidebar

## What type of PR is this? (check all applicable)


## Description

While we now support generalized profile fields (including a dynamically generated page for user's to update their profiles), we are not really using this information when displaying the profile, as was most recently pointed out by @AfroDevGirl [here](https://github.com/forem/forem/issues/6994#issuecomment-716571684)

As a quickfix, this PR makes the left sidebar of the profile dynamic, by rendering the fields with `display_area: :left_sidebar` there. This is **not** a complete overhaul of the profile page, only the sidebar, which is a step in the right direction. Maybe someone else can help out with the header, which I left untouched for now.

I also used Blazer to check the current profile fields on DEV, and 🎉 , all of them are assigned to `:left_sidebar`, which is definitely not what we want. So this PR also includes a data update script to assing all of the existing DEV fields to the correct areas and adds a new area, for fields that are only needed as settings. Two related questions:

1. Can we still do this via a data update script or will this negatively impact Forems which already modified their profiles? I can do a temporary Rake task instead that we run for DEV and that other communities can run if they choose too? CC @benhalpern 
2. Does it even make sense to have profile fields that are only used as settings? DId we over-generalize here? If they are configurable, we can't rely on them being there, which makes them not very useful as settings (e.g. the two color fields). Should these remain on `User` instead? That said, they currently also include things like `display_looking_for_work_on_profile` or `recruiters_can_contact_me_about_job_opportunities`, which are pretty DEV specific and might not apply to other communities at all.

## Related Tickets & Documents

Part of #6994

## QA Instructions, Screenshots, Recordings

1. Make sure you have some profile fields that are have a display area of `:left_sidebar`.
2. Run the code and make sure the profile page renders as you'd expect it to.

## Added tests?

- [X] Yes, but it's not working yet, so back to draft.

## Added to documentation?


## [optional] Are there any post deployment tasks we need to perform?

## [optional] What gif best describes this PR or how it makes you feel?

![image](https://user-images.githubusercontent.com/47985/97254406-e1312500-1840-11eb-93cb-71961979c9e1.png)
